evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -52,6 +54,18 @@ def next_batch(self):
         if self._batch_number == 0:
             next_batch = [pipeline_class(parameters=self._transform_parameters(pipeline_class, {}))
                           for pipeline_class in self.allowed_pipelines]
+
+        # One after training all pipelines one round
+        elif (len(self._first_batch_results) > 1 and
+              self._batch_number != 1 and
+              (self._batch_number) % (len(self._first_batch_results) + 1) == 0):
+            input_pipelines = []
+            for i in range(len(self._first_batch_results)):
+                pipeline_class = self._first_batch_results[i][1]
```

Comment:
@angela97lin so this ensembler is running after `1 + len(self._first_batch_results)` batches, correct? Where the "1" is for the first batch, and then we run `len(self._first_batch_results)` more batches to do deeper tuning on specific pipelines.

The reason I ask is, it appears we're allowing that number of batches to pass before adding the stacked ensembler. But, the `input_pipelines` list is constructed using only the pipelines from the first batch.

I think we should either a) run stacking immediately after the first batch, using the input list you've got here, or b) run stacking after `1 + len(self._first_batch_results)` batches, using an input list with the latest results. Both are cool starting points, but given that stacking takes a while to run, I'm a fan of option b, because it'll occur later in the process, and theoretically will also be more accurate using the non-first-batch pipelines!

Regarding how to implement that: we haven't had to do this kind of operation in the automl algo to date! I think we'll need a new datastructure (another list like `_first_batch_results`? something else) to make this easy to do. Let's chat about it! :)